gateway/ln-gateway/src/state_machine/pay.rs

```diff
@@ -164,7 +168,7 @@ impl Display for OutgoingPaymentError {
 
 #[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
 pub struct GatewayPayInvoice {
-    pub contract_id: ContractId,
+    pub pay_invoice_payload: PayInvoicePayload,
```

Comment:
thought: it's slightly odd to see the payload abstraction being passed through to state machines

Reply:
It seemed to me to be the most convenient way to pass around this data. I didn't think making a new struct was worth it, but I'm open to suggestions